### DISCUSSION TITLE
Possible spam resistance system using the hashcash proof of work system

### DISCUSSION THREAD

**Discussion by Destroyinator69420** [freenet/freenet-core#137] [General]
Since Locutus will be processing messages, it will be necessary to prevent spam and DOS attacks. I think that hashcash serve this purpose. It is proven, being used in Bitcoin as well as email spam filters. If spam and DOS attacks are not accounted for, than these attacks can push aside legitimate users who wish to use the network. By adding a small cost to sending a message, than it would be much harder to send large amounts of spam, since a small amount of computing power would be needed per message, this would not be much of a burden on legitimate users, but it would make spam operations and DOS attacks exponentially more computationally expensive.

All the best,
Destroyer

**Comment by iduartgomez** [comment]
We intend to check resource usage on behave of a peer by the different peers in the network, that will be tied to some sort of karma, currency, etc. system that is not yet well defined but we are gonna have this sort of systems to try prevent DOS attacks and make them difficult and is definitively in our minds.

**Comment by micwin** [comment]
+1 - when defining spam as "low importance traffic from low karma senders",  priorizing messages proportionally to its senders' karma should stop spam right in the making :-)

That said - if the karma system is hard enough - spam is done. What a glorious future!
